miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/AuthorizationManager.java

```diff
@@ -92,16 +108,35 @@
    */
   public boolean writeCheck(SecurableByProfile resource) throws IOException;
   
+  /**
+   * Checks whether the provided user has write permission on a resource
+   * 
+   * @param resource the object on which to check permissions
+   * @param user the User for whom to check permission
+   * @return true if the user has permission to modify this object; false otherwise
+   * @throws NullPointerException if resource is null
+   */
+  public boolean writeCheck(SecurableByProfile resource, User user);
+
   /**
    * Verifies that the current user has permission to modify a resource and throws an AuthorizationException if not
    * 
    * @param resource the object to check permissions for
    * @throws AuthorizationException if the current user does not have write permission to the resource
-   * @throws IOException if there is an error while looking up the current user. This does not indicate whether 
-   * the user exists or is authenticated, but represents other errors occurring while checking these conditions
+   * @throws IOException if there is an error while looking up the current user. This does not indicate whether
+   *           the user exists or is authenticated, but represents other errors occurring while checking these conditions
    */
   public void throwIfNotWritable(SecurableByProfile resource) throws IOException, AuthorizationException;
   
+  /**
+   * Verifies that the provided user has permission to modify a resource and throws an AuthorizationException if not
+   * 
+   * @param resource the object on which to check permissions
+   * @param user the User for whom to check permissions
+   * @throws AuthorizationException if the current user does not have write permission to the resource
+   */
+  public void throwIfNotWritable(SecurableByProfile resource, User user) throws AuthorizationException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'uk.ac.bbsrc.tgac.miso.service.security.AuthorizationException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)